tests/admin_inlines/models.py

```diff
@@ -326,6 +326,36 @@ class Meta:
         proxy = True
 
 
+# Models for #31867
+class Parent1(models.Model):
+    name = models.CharField(max_length=1)
+
+
+class Child1(models.Model):
+    name = models.CharField(max_length=1)
+    position = models.PositiveIntegerField()
+    parent = models.ForeignKey(Parent1, models.CASCADE)
+
+
+class Parent2(models.Model):
+    name = models.CharField(max_length=1)
+
+
+class Child2(models.Model):
+    name = models.CharField(max_length=1)
+    position = models.PositiveIntegerField()
+    parent = models.ForeignKey(Parent2, models.CASCADE)
+
+
+class Parent3(models.Model):
+    name = models.CharField(max_length=1)
```

Comment:
Do we need so many new models? e.g. it should be enough to have a single "Parent" model and register it with 3 inlines. We could also use better names rather than `Child1/2/3`, e.g. `ChildHiddenFieldsTabular`.